File: 20A40/cir-Israel/Ejercicio 40/CSR.cpp
#pragma once
#include "Manager.h"

class CSR : public Employee {
private:
    std::string headSeType;
    Manager myMgr;
    std::string loginID;

public:
    // Constructor
    CSR(std::string name, std::string address, float salary, 
        std::string headSeType, Manager myMgr, std::string loginID);

    void setHeadSeType(std::string headSeType);
    void setMyMgr(Manager myMgr);
    void setLoginID(std::string loginID);

    std::string getHeadSeType() const;
    Manager getMyMgr() const;
    std::string getLoginID() const;

    // Otros métodos
    void clocksIn();
    void clockOut();

    std::string toString() const;
};

File: 20A40/cir-villalobos/completos/e40/CSR.java
public class CSR extends Employee {

    private String headSetType;
    private Manager myMgr;
    private String loginID;

    public boolean setHeadSetType(String typ){
        if(!(typ.length() > 0))
            return false;
        this.headSetType = typ;
        return true;
    }

    public boolean setLoginID(String id){
        if(!(id.length() > 0))
            return false;
        this.loginID = id;
        return true;
    }

    public boolean setManager(Manager mgr){
        if(!(mgr != null))
            return false;
        this.myMgr = mgr;
    }


    public String getHeadSetType() {    return this.headSetType;  }
    public String getLoginID()     {    return this.loginID;      }
    public Manager getMyMgr()      {    return this.myMgr;        }

    @Override
    public String toString(){
        String concat = super.toString() + "\tHeadset type: " + headSetType +
                 "\tLogin ID: " + loginID + "\tManager: " + myMgr.getName();
        return concat;
    }
}
